src/components/VideoPlayer/index.js

```diff
@@ -0,0 +1,184 @@
+  };
+
+  const fullScreen = () => {
+    if (videoRef.current.requestFullscreen) {
+      videoRef.current.requestFullscreen();
+    } else if (videoRef.current.webkitRequestFullscreen) {
+      /* Safari */
+      videoRef.current.webkitRequestFullscreen();
+    } else if (videoRef.current.msRequestFullscreen) {
+      /* IE11 */
+      videoRef.current.msRequestFullscreen();
+    }
+  };
+
+  useEffect(() => {
+    if (state.play) videoRef.current.play();
+    else videoRef.current.pause();
+  }, [state.play]);
+
+  return (
+    <div className={classes.videoPlayer}>
+      <video
+        src={src}
+        className={classes.video}
+        ref={videoRef}
+        onClick={togglePlay}
+        onTimeUpdate={timeUpdateHandler}
+        muted={state.play ? "" : "muted"}
+      ></video>
+
+      <div className={classes.controller}>
+        <div
+          className={classes.progress}
+          ref={progressBar}
+          onClick={clickToSkip}
+        >
+          <div
+            className={classes.progress_filled}
+            style={{
+              width: `${progress}%`,
+            }}
+          >
+            <i className="fas fa-circle"></i>
+          </div>
+        </div>
+
+        <div className={classes.controller_options}>
+          <button className={classes.play} onClick={togglePlay}>
+            {state.play ? (
+              <i className="fas fa-pause"></i>
+            ) : (
+              <i className="fas fa-play"></i>
+            )}
+          </button>
+          <input
+            type="range"
+            min={0}
+            max={1}
+            step={0.05}
+            className={classes.volume}
+            defaultValue={0.5}
+            onChange={(e) => (videoRef.current.volume = e.target.value)}
+          />
+
+          {/* playback for the video starts here */}
+          <div className={classes.playback}>
+            <i
+              onClick={() => dispatch({ type: "PLAYBACK" })}
+              className={"fas fa-tachometer-alt" + " " + classes.playback_icon}
+            ></i>
+            <div
+              className={
+                state.playback
+                  ? classes.playback_dropdown +
+                    " " +
+                    classes.playback_dropdownActive
+                  : classes.playback_dropdown
+              }
+            >
+              <button onClick={playbackRateHandler} speed={0.25}>
+                0.25
+              </button>
+              <button onClick={playbackRateHandler} speed={0.5}>
+                0.5
+              </button>
+              <button onClick={playbackRateHandler} speed={1}>
+                Normal
+              </button>
+              <button onClick={playbackRateHandler} speed={1.25}>
+                1.25
+              </button>
+              <button onClick={playbackRateHandler} speed={1.5}>
+                1.5
+              </button>
+              <button onClick={playbackRateHandler} speed={2}>
```

Comment:
Appreciation: You have implemented playback speed too. This is clean great work.